src/Amazon.Extensions.CognitoAuthentication/CognitoUserAuthentication.cs

```diff
@@ -55,6 +55,8 @@ public virtual async Task<AuthFlowResponse> StartWithSrpAuthAsync(InitiateSrpAut
             RespondToAuthChallengeRequest challengeRequest =
                 CreateSrpPasswordVerifierAuthRequest(initiateResponse, srpRequest.Password, tupleAa);
 
+            challengeRequest.ClientMetadata = new Dictionary<string, string>(srpRequest.ClientMetadata);
```

Comment:
Can you add a null check for `srpRequest.ClientMetadata` before running this line. In my testing since `srpRequest.ClientMetadata` was null the `new Dictionary` call through a null pointer exception.

Reply:
Hey @normj, thanks for taking a look at this. Added a conditional to avoid the null pointer exception.